include/seqan3/contrib/parallel/buffer_queue.hpp

```diff
@@ -281,47 +360,53 @@ inline bool buffer_queue<value_t, buffer_t, buffer_policy>::overflow(value2_t &&
 {
     // try to extend capacity
     std::unique_lock write_lock{mutex};
-    size_type cap = data.size();
-    size_type roundSize = this->roundSize;
-    size_type headPos = this->headPos;
-    size_type tailPos = this->tailPos;
 
-    assert(tailPos == this->tailWritePos);
-    assert(headPos == this->headReadPos);
+    size_type old_size = buffer.size();
+    size_type ring_buffer_capacity = this->ring_buffer_capacity;
+    size_type pop_front_position = this->pop_front_position;
+    size_type push_back_position = this->push_back_position;
+
+    // Expects no pending pushes or pops in unique lock.
+    assert(push_back_position == this->pending_push_back_position);
+    assert(pop_front_position == this->pending_pop_front_position);
 
     bool valueWasAppended = false;
 
     // did we reach the capacity limit (another thread could have done the upgrade already)?
-    if (cyclic_increment(tailPos, cap, roundSize) >= headPos + roundSize)
+    // buffer is full if tail_pos + 1 == head_pos
+    if (!has_free_slots(pop_front_position, cyclic_increment(push_back_position)))
```

Comment:
use the `is_full` member function?

Reply:
Well, that wouldn't be possible since a) is_full also acquires a unique lock and you can;t do that b) the question is if it is full after the push_back_position was incremented by one. That is not the semantic of the is_full member function.